src/codeu/chat/util/GUI.java

```diff
@@ -0,0 +1,425 @@
+
+
+        //Conversation panel holding conversation panel's GUI, same as root panel but different buttons
+        JPanel convoPanel = new JPanel();
+        convoPanel.setLayout(new GridBagLayout());
+        GridBagConstraints convoConstraints = new GridBagConstraints();
+        convoConstraints.fill = GridBagConstraints.BOTH;
+        switchPanel.add(convoPanel, "convoPanel");
+//		panelSwitcher.show(switchPanel, "convoPanel");
+
+        //message & text display window in convo panel, should display any messages/output while in convo panel
+        JTextArea messages = new JTextArea("Application Activity:\nType in input text box and press a command button.", 20, 20);
+        messages.setLineWrap(true);
+        messages.setLayout(new FlowLayout());
+        PrintStream convoOutput = new PrintStream(new GUIOutputDisplay(messages));
+        JScrollPane convoScroll = new JScrollPane(messages);
+
+        JButton addConvo = new JButton("Add Chat");
+        addConvo.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                chat.handleCommand("c-add " + userTextInput.getText());
+                System.setOut(convoOutput);
+                System.setErr(convoOutput);
+                panelSwitcher.show(switchPanel, "convoPanel");
+            }
+        });
+        userButtonPanel.add(addConvo);
+        JButton joinConvo = new JButton("Join Chat");
+        joinConvo.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent arg0) {
+                chat.handleCommand("c-join " + userTextInput.getText());
+                System.setOut(convoOutput);
+                System.setErr(convoOutput);
+                panelSwitcher.show(switchPanel, "convoPanel");
+            }
+
+        });
+        userButtonPanel.add(joinConvo);
+
+        convoConstraints.gridx = 1;
+        convoConstraints.gridy = 0;
+        convoConstraints.gridwidth = 2;
+        convoConstraints.gridheight = 2;
+        convoPanel.add(convoScroll, convoConstraints);
+
+        //Panel with command buttons and text input
+        JPanel convoInputPanel = new JPanel();
+        convoInputPanel.setLayout(new GridLayout(2, 1));
+        convoConstraints.gridx = 1;
+        convoConstraints.gridy = 2;
+        convoConstraints.gridwidth = 2;
+        convoConstraints.gridheight = 1;
+        convoPanel.add(convoInputPanel, convoConstraints);
+
+        //Text input for user's messages/other conversation commands
+        JTextField convoTextInput = new JTextField(1);
+        convoInputPanel.add(convoTextInput);
+
+        //Command buttons
+        JPanel convoButtonPanel = new JPanel();
+        convoButtonPanel.setLayout(new GridLayout(1, 9));
+        convoInputPanel.add(convoButtonPanel);
+        JButton convoHelp = new JButton("Help");
+        convoHelp.addActionListener(new ActionListener() {
```

Comment:
Code has been simplified. I tried incorporating the output stream changes and panel switching into the `createActionButton` method's ActionListener for the buttons that need to trigger those changes, but it couldn't be done without making a bunch more field variables, so I just left those JButton creation sessions as is.